model/project_ref.go

```diff
@@ -325,9 +325,29 @@ func (projectRef *ProjectRef) Insert() error {
 
 func (p *ProjectRef) Add(creator *user.DBUser) error {
 	p.Id = mgobson.NewObjectId().Hex()
+
+	// if a hidden project exists for this configuration, use that ID
+	if p.Owner != "" && p.Repo != "" && p.Branch != "" {
+		hidden, err := FindHiddenProjectRefByOwnerRepoAndBranch(p.Owner, p.Repo, p.Branch)
+		if err != nil {
+			return errors.Wrapf(err, "error querying for hidden project")
+		}
+		if hidden != nil {
+			p.Id = hidden.Id
+			err := p.Upsert()
```

Comment:
Does it make sense to unhide the project here?

Reply:
Assuming the project p wasn't explicitly marked Hidden: true, then it will be unhidden by doing the upsert.